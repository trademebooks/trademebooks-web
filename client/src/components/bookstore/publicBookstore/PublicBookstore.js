import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import { MDBRow, MDBCol, MDBContainer } from 'mdbreact'
import Books from '../../books/common/Books'
import api from '../../../utils/api';

const PublicBookstore = ({ match, books, getBookstoreByUsername }) => {
  const [formData, setFormData] = useState({
    location: '',
    school: ''
  })

  const { location, school } = formData

  useEffect(() => {
    const fetchAccount = async() => {
      const res = await api.get("/account")
      const { location, school } = res.data.data
      setFormData({location: location, school: school})
    }
    fetchAccount()
    getBookstoreByUsername(match.params.username)
  }, [match.params.username, getBookstoreByUsername])

  return (
    <>
        <div className="mt-4">
          <MDBContainer>
            <MDBRow>
              <MDBCol sm="12">
                <h3 className="font-weight-bold">
                  Welcome to {match.params.username}'s Bookstore
                </h3>
              </MDBCol>
            </MDBRow>
          </MDBContainer>

          <MDBContainer>
            <MDBRow>
                <div className="col-xs-6 ml-4">
                  <i className="fas fa-map-marker-alt" style={{fontSize: "300%", height: "150%", color: "var(--primary-dark-color)"}}></i>
                </div>
                <div className="col-xs-12">
                    Located in:<br />
                    <strong>{location}</strong>
                </div>

                <div className="col-xs-6 ml-4">
                  <i className="fas fa-graduation-cap" style={{fontSize: "300%", height: "150%", color: "var(--primary-dark-color)"}}></i> 
                </div>
                <div className="col-xs-12">
                    Selling books for:<br />
                    <strong>{school}</strong>
                </div>
            </MDBRow>
          </MDBContainer>
        </div>
        
        <div className="mt-4">
          <MDBContainer>
            <MDBRow center={true}>
              <MDBCol sm="12">
                {books.length < 1 ? (
                  <div>There are no books in this bookstore...</div>
                ) : (
                  <Books books={books} />
                )}
              </MDBCol>
            </MDBRow>
          </MDBContainer>
        </div>
    </>
  )
}

PublicBookstore.propTypes = {
  match: PropTypes.object.isRequired,
  books: PropTypes.array.isRequired,
  getBookstoreByUsername: PropTypes.func.isRequired
}

export default PublicBookstore
