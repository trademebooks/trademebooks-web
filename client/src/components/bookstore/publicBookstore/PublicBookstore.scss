.icon--style {
  font-size: 2rem;
  color: var(--primary-dark-color);
}

.xyz {
  display: inline !important;
  width: auto !important;
}
