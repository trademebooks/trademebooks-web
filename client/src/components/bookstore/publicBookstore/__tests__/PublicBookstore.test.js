import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'

import PublicBookstore from '../PublicBookstore'

describe('Component: PublicBookstore', () => {
  const props = {
    match: {
      params: {
        username: 'johnny123'
      }
    },
    books: [],
    getBookstoreByUsername: jest.fn()
  }

  it('should match the snapshot', () => {
    const publicBookstoreSnapshot = renderer
      .create(<PublicBookstore {...props} />)
      .toJSON()
    expect(publicBookstoreSnapshot).toMatchSnapshot()
  })

  it('should render properly', () => {
    render(<PublicBookstore {...props} />)
  })
})
