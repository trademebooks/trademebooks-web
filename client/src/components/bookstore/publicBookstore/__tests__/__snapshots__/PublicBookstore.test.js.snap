// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PublicBookstore should match the snapshot 1`] = `
Array [
  <div
    className="mt-4"
  >
    <div
      className="container"
      data-test="container"
    >
      <div
        className="row"
        data-test="row"
      >
        <div
          className="col-sm-12"
          data-test="col"
        >
          <h3>
            Welcome to
             
            <span
              className="font-weight-bold"
            >
              johnny123
            </span>
            's Bookstore
          </h3>
        </div>
      </div>
    </div>
    <div
      className="container"
      data-test="container"
    >
      <div
        className="row"
        data-test="row"
      >
        
        
      </div>
    </div>
  </div>,
  <div
    className="mt-4"
  >
    <div
      className="container"
      data-test="container"
    >
      <div
        className="row justify-content-center"
        data-test="row"
      >
        <div
          className="col-sm-12"
          data-test="col"
        >
          <div>
            There are no books in this bookstore...
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
